bakkesmod-playlist-rank-viewer/bakkesmod-playlist-rank-viewer.cpp

```diff
@@ -76,47 +98,123 @@ void PlaylistRankViewer::timeoutCallback(GameWrapper* _) {
+	}
+}
+
+// Update the MMR/Rank data and map for the user if it isnt updated yet
+void PlaylistRankViewer::updatePlayerStats(UniqueIDWrapper uniqueId) {
 	MMRWrapper mmrWrapper = gameWrapper->GetMMRWrapper();
-	long long steamId = id.ID;
-	if (playerMmrs.count(steamId) == 0 || playerMmrs[steamId].size() == 0) {
-		playerMmrs[steamId] = {};
+
+	long long uID = uniqueId.GetUID();
+
+	if (playerStats.count(uID) == 0 || playerStats[uID].size() == 0) {
+		playerStats[uID] = {};
+		bool showRank = cvarManager->getCvar("PRV_show_rank").getBoolValue();
+		bool showMmr = cvarManager->getCvar("PRV_show_mmr").getBoolValue();
+		bool showGamesPlayed = cvarManager->getCvar("PRV_show_games_played").getBoolValue();
+
 		for (auto playlist : playlistsToCheck) {
-			string mmr = to_string(mmrWrapper.GetPlayerMMR(id, playlist));
-			string gamesPlayed = to_string(mmrWrapper.GetPlayerRank(id, playlist).MatchesPlayed);
-			playerMmrs[steamId][playlist] = mmr + " (" + gamesPlayed + ")";
+			playerStats[uID][playlist] = {};
+
+			SkillRank userRank = mmrWrapper.GetPlayerRank(uniqueId, playlist);
+			string mmr = to_string_with_precision(mmrWrapper.GetPlayerMMR(uniqueId, playlist), 1);
+			string gamesPlayed = to_string(userRank.MatchesPlayed);
+			string currentRankName = GetRankName(userRank.Tier, userRank.Division);
+			Color rankColor = GetRankColor(userRank.Tier);
+
+			playerStats[uID][playlist].color = rankColor;
+
+			if (playlist != PLAYLIST::UNRANKED && showRank) {
+				playerStats[uID][playlist].text = currentRankName + " ";
+			}
+
+			if (showMmr) {
+				playerStats[uID][playlist].text += mmr + " ";
+			}
+
+			if (showGamesPlayed) {
+				playerStats[uID][playlist].text += "(" + gamesPlayed + ")";
+			}
 		}
 	}
 }
 
-void PlaylistRankViewer::writeStats(CanvasWrapper& canvas, long long uniqueId, string playerName) {
-	Vector2 screen = canvas.GetSize();
-	
-	float nameX = screen.X * .85;
-	float nameY = screen.Y * .1;
-	float tableX = nameX;
-	float tableY = nameY + 50.0;
+void PlaylistRankViewer::writeStats(CanvasWrapper& canvas, UniqueIDWrapper uniqueId, string playerName) {
+	float PRV_x_position = cvarManager->getCvar("PRV_x_position").getFloatValue();
+	float PRV_y_position = cvarManager->getCvar("PRV_y_position").getFloatValue();
+	float PRV_scale = cvarManager->getCvar("PRV_scale").getFloatValue();
+	int PRV_background_opacity = cvarManager->getCvar("PRV_background_opacity").getIntValue();
+
+	long long uID = uniqueId.GetUID();
 
-	// Draw a box for the stats to go into
-	canvas.SetColor(100, 100, 100, 150);
-	canvas.SetPosition(Vector2{int(nameX - 10), int(nameY - 10)});
-	// Estimating the required width and height of the box
-	canvas.FillBox(Vector2{int(240), int(70 + (20 * playlistsToCheck.size()))});
+	// Draw box here
+	Vector2 drawLoc = {
+		int(canvas.GetSize().X * PRV_x_position),
+		int(canvas.GetSize().Y * PRV_y_position)
+	};
+	Vector2 sizeBox = {
+		int(255 * PRV_scale),
+		int(int(35 + (20 * playlistsToCheck.size())) * PRV_scale)
```

Comment:
Could we change 35 -> 40? Its cut a little close for my tastes

Reply:
Of course!